avm/res/logic/integration-account/agreement/main.bicep

```diff
@@ -0,0 +1,85 @@
+metadata name = 'Integration Account Agreements'
+metadata description = 'This module deploys an Integration Account Agreement.'
+
+// =============== //
+//   Parameters    //
+// =============== //
+
+@description('Required. The Name of the agreement resource.')
+param name string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Conditional. The name of the parent integration account. Required if the template is used in a standalone deployment.')
+param integrationAccountName string
+
+@description('Required. The agreement type.')
+param agreementType ('AS2' | 'Edifact' | 'NotSpecified' | 'X12')
+
+@description('Required. The business identity of the guest partner.')
+param guestIdentity businessIdentityType
+
+@description('Required. The integration account partner that is set as guest partner for this agreement.')
+param guestPartner string
+
+@description('Required. The business identity of the host partner.')
+param hostIdentity businessIdentityType
```

Comment:
Why a custom and not a resource-derived type if only the standard properties are used?

Reply:
Changed to resource-derived type